RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -325,39 +325,35 @@ class PumpOpsSynchronous {
         return results
     }
     
-    internal func getHistoryEventsSinceDate(startDate: NSDate) throws -> ([PumpEvent], PumpModel) {
-        
+    internal func getHistoryEventsSinceDate(startDate: NSDate) throws -> ([TimestampedHistoryEvent], PumpModel) {
         try wakeup()
         
         let pumpModel = try getPumpModel()
-        
-        var pageNum = 0
-        var events = [PumpEvent]()
-        while pageNum < 16 {
+
+        var events = [TimestampedHistoryEvent]()
+
+        pages: for pageNum in 0..<16 {
```

Comment:
Hahahaha I suppose you're right! This was the only use case in 2 years I've found for them.